### PR TITLE
Make dashboard shortcuts configurable

### PR DESCRIPTION
**Pull request by winny-**
- Makes shortcuts configurable
- Only show shortcuts that are not `nil`
- Use cleaner identifiers for the shortcut lambdas
- Add docstrings to the shortcut lambda (visible with e.g. `C-h k p`)

I'm an elisp novice, any suggestions appreciated :)

### CODE REVIEW

**Comment by winny-** [comment]
Something I wasn't able to get working quite right is: `C-h f dashboard-jump-to-recent-files RET`. Emacs says `user-error: Symbol’s function definition is void: dashboard-jump-to-recent-files`.

I don't think it is completely necessary to be able to get this working, though it would be what the user expects. Any suggestions appreciated :)

File: dashboard-widgets.el [hunk 1]
```diff
@@ -254,15 +265,30 @@ Return entire list if `END' is omitted."
...
   `(progn
      (eval-when-compile (defvar dashboard-mode-map))
-     (let ((sym (make-symbol (format "Jump to \"%s\"" ,search-label))))
+     (let ((sym (make-symbol (format "dashboard-jump-to-%s" (downcase
+                                                             (replace-regexp-in-string
+                                                              " +" "-" (replace-regexp-in-string
+                                                                        ":+" "" ,search-label
+                                                                        nil nil nil)
+                                                              nil nil nil))))))
        (fset sym (lambda ()
```
Comment (kisaragi-hiu) [inline dashboard-widgets.el]: This function definition happens within the `let` during runtime—it is thrown away after the runtime `let` exits. This is why `C-h f dashboard-jump-to-recent-files` fails. This needs to be replaced with a `defun`, like:

```elisp
  (let ((sym (make-symbol (format "dashboard-jump-to-%s" (downcase search-label)))))
    `(progn
       (eval-when-compile (defvar dashboard-mode-map))
       (defun ,sym nil
         ,(concat
           "Jump to "
           (replace-regexp-in-string
            ":+" "" (format "%s" search-label) nil nil nil)
           ".  This code is dynamically generated in `dashboard-insert-shortcut'.")
         (interactive)
         (unless (search-forward ,search-label (point-max) t)
           (search-backward ,search-label (point-min) t))
         ,@(unless no-next-line
             '((forward-line 1)))
         (back-to-indentation))
```

Comment (winny-) [reply dashboard-widgets.el]: Thanks for the suggestion! I tried this out in another branch, but was not able to get it working. The code works, but indeed I am not able to do either `M-x dashboard-jump-to-projects RET` or `C-h f dashboard-jump-to-project RET` (with the same error as above).

https://github.com/winny-/emacs-dashboard/commit/0e0a56da0b19fbb00a8618119d3ed9d604b0de0c

Comment (kisaragi-hiu) [reply dashboard-widgets.el]: Looking at your linked commit, the problem is that you're using `make-symbol`, which gives you an uninterned symbol, ie. one that is unique. You need to use `intern` instead.

```elisp
(eq 'testing (intern "testing")) ; => t
(eq 'testing (make-symbol "testing")) ; => nil
```

Comment (winny-) [reply dashboard-widgets.el]: Thank you very much for the help! That did the trick! =)

**Comment by winny-** [comment]
Applied the changes suggested, let me know if there are any other things to adjust =)

**Comment by winny-** [comment]
Hello. Is there interest to merge this feature in? Does this need further review?

**Comment by jcs090218** [comment]
This is neat. I like the idea. I don't think merging this would have any issue.

I will make a quick test next week. Thanks for the PR!

**Comment by jcs090218** [comment]
I have tested this PR and it works well!

Once thing I might worry is that the other extension that extends this package might want to know this; yet there is no conflict to this.

**Review by jcs090218 — APPROVED** [review]

**Comment by jcs090218** [comment]
Merged! Thank you!

**Comment by winny-** [comment]
Cheers.  Congrats on the project transition!

**Pull request merged**